source/ci_source/providers/GitHubActions.ts

```diff
@@ -161,21 +163,24 @@ export class GitHubActions implements CISource {
   }
 
   get useEventDSL() {
-    // Support event based PR runs
-    return this.env.GITHUB_EVENT_NAME !== "pull_request"
+    return this.event.pull_request !== undefined || this.event.issue !== undefined
   }
 
   get pullRequestID(): string {
-    if (this.env.GITHUB_EVENT_NAME === "pull_request") {
+    if (this.event.pull_request !== undefined) {
       return this.event.pull_request.number
+    } else if (this.event.issue !== undefined) {
+      return this.event.issue.number
```

Comment:
This part is a little bit controversial, I've added it because technically the PR is a particular type of issue, and then I was thinking to someone that wants to run danger when a pr comment is added, and in that case I think he would have to use `issue_comment`, `pull_request_review` and `pull_request_review_comment` triggers.
In issue `comment` there is no pull request JSON, only the issue one https://developer.github.com/v3/activity/events/types/#webhook-payload-example-14

Reply:
Happy to remove it if is a bad idea

Reply:
I'm open to seeing where this goes 👍